### PR TITLE
update ocaml-native to 4.02.3

### PR DESCRIPTION
**Pull request by starius**
See https://lists.nongnu.org/archive/html/mingw-cross-env-list/2015-08/msg00003.html

### CODE REVIEW

File: src/ocaml-native-1-fixes.patch [hunk 1]
```diff
@@ -24,78 +27,70 @@ index b5ef878..77454ed 100644
...
-+	   (ocamlmktop,"ocamlmktop")]
-   end;
- 
-   let reorder x y = x := !x @ (List.concat (List.rev !y)) in
+@@ -232,7 +235,7 @@ let spec = ref (
+    "-install-lib-dir", Set_string Ocamlbuild_where.libdir, "<path> Set the install library directory";
+    "-install-bin-dir", Set_string Ocamlbuild_where.bindir, "<path> Set the install binary directory";
+    "-where", Unit (fun () -> print_endline !Ocamlbuild_where.libdir; raise Exit_OK), " Display the install library directory";
+-   "-which", String (fun cmd -> print_endline (find_tool cmd); raise Exit_OK), "<command> Display path to the tool command";
++   "-which", String (fun cmd -> print_endline (find_tool cmd "FIXME"); raise Exit_OK), "<command> Display path to the tool command";
```
Comment (starius) [inline src/ocaml-native-1-fixes.patch]: Please review this line carefully. Is it OK to replace extra argument with string "FIXME"?

**Comment by starius** [comment]
Package ocaml-core must be updated as well. It shares ocaml version with ocaml-native.

File: src/ocaml-native-1-fixes.patch (hunk 1 above)
Comment (tonytheodore) [inline src/ocaml-native-1-fixes.patch]: I think so, I don't know anything about ocaml but an updated version with minor issues is better than an outdated version.

**Comment by tonytheodore** [comment]
From #1003, this should also be moved to the native target so it's built only once. It's [building on OSX now](https://github.com/mxe/mxe/commit/a96fcf41143baa5aaf3c0e31a9fe204089fad3d9) so will be easier for me to test.

**Comment by tonytheodore** [comment]
Also, project has moved:

http://ocaml.org/
https://github.com/ocaml/ocaml

I wonder if we should make this a plugin?

P.S. @starius, could you update the checksum on this and rebase?

**Comment by starius** [comment]
Rebased, updated checksum, patch consolidated.

It doesn't work. I failed to update crosscompiled ocaml. Somebody who knows ocaml should look at it.

> I wonder if we should make this a plugin?

Why?

**Comment by tonytheodore** [comment]
See #2512

**Pull request closed without merging**